### PR TITLE
Fix table data px dimension parsing bug

### PR DESCRIPTION
**Pull request by HrsLizo**
e.g. Fixes #0 

**Checklist**:

- [ ] A unit test is covering the code added / modified by this PR N/A

- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder N/A

- [ ] A mention of the change is present in `CHANGELOG.md` N/A

- [x] This PR is ready to be merged 

Added a .replace("px", "") strip to safely handle pixel dimensions during HTML table parsing.

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: fpdf/html.py [hunk 1]
```diff
@@ -1125,6 +1125,9 @@ def handle_starttag(self, tag: str, attrs: list[tuple[str, str | None]]) -> None
             if not self.table or not self.table_row:
                 raise FPDFException(f"Invalid HTML: <{tag}> used outside any <tr>")
             self.td_th = {k.lower(): v for k, v in attrs_dict.items()}
+            for dimension in ("width", "height"):
+            if dimension in self.td_th and self.td_th[dimension] is not None:
+                self.td_th[dimension] = self.td_th[dimension].replace("px", "").strip()
```
Comment (andersonhc) [inline fpdf/html.py:1130]:
```suggestion
            for dimension in ("width", "height"):
                if dimension in self.td_th and self.td_th[dimension] is not None:
                    self.td_th[dimension] = self.td_th[dimension].replace("px", "").strip()
```

**Review by andersonhc — CHANGES_REQUESTED** [review]
Changes requested:
- Fix indentation
- Add a test in `test/html/test_html.py` exercising your change
- Add a CHANGELOG entry

If you need clarification or assistance in any of these please let us know.

**Review by Pawansingh3889 — COMMENTED** [review]
Good catch on stripping  from HTML table dimensions. Two observations:

1. **Indentation issue**: The  and  lines need to be indented one level deeper to be inside the  loop:



As written, only the  line runs in the loop, and the / runs once with  set to  (the last value).

2. **Other units**: HTML also supports , , ,  in dimension attributes. You might want to handle those too, or at minimum strip non-numeric characters:



Otherwise the approach is solid. The  stripping is the most common case users will hit.

**Review by Pawansingh3889 — COMMENTED** [review]
Thanks for tackling this — I hit similar dimension parsing issues when generating PDF audit reports from HTML tables.

I notice an indentation issue in the diff: the `if dimension in self.td_th` line appears to be at the same indent level as the `for` loop rather than inside it. This would cause a syntax error or incorrect behavior — the `if` block needs to be indented one level deeper under the `for dimension in ...` loop.

Also worth considering:
- Should this also handle other CSS units like `em`, `rem`, `pt`, `%`? At minimum a regex like `re.sub(r"[a-z%]+$", "", value)` would be more robust.
- Edge case: what happens with `width="auto"`? The `.replace("px", "")` would pass it through unchanged and it may fail downstream when parsed as a number.

A test case for HTML like `<td width="120px">` would help confirm the fix works end-to-end.

**Comment by andersonhc** [comment]
Hi @HrsLizo 

Just checking in to see if you saw and the review and if you're still interested in completing this PR?
